modules/account-lib/src/coin/algo/transaction.ts

```diff
@@ -210,14 +206,22 @@ export class Transaction extends BaseTransaction {
       tokenId: this._algoTransaction?.assetIndex,
       genesisID: this._algoTransaction.genesisID,
       genesisHash: this._algoTransaction.genesisHash.toString('base64'),
+      txType: TransactionType[this.type],
+      amount: '0',
     };
+
     if (this._algoTransaction.closeRemainderTo) {
       result.closeRemainderTo = algosdk.encodeAddress(this._algoTransaction.closeRemainderTo.publicKey);
     }
-    if (this.type === TransactionType.Send) {
+
+    if (this._algoTransaction.to) {
       result.to = algosdk.encodeAddress(this._algoTransaction.to.publicKey);
+    }
```

Comment:
you could do something like this instead the if statement, if you like.
result.to = algosdk.encodeAddress(this._algoTransaction.to?.publicKey);